airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -161,6 +161,23 @@ def get_client(self, project_id: Optional[str] = None, location: Optional[str] =
             credentials=self._get_credentials(),
         )
 
+    def get_uri(self) -> str:
+        """Override DbApiHook get_uri method for get_sqlalchemy_engine()"""
+        return "bigquery://"
+
+    def get_sqlalchemy_engine(self, engine_kwargs=None):
+        """
+        Get an sqlalchemy_engine object.
+
+        :param engine_kwargs: Kwargs used in :func:`~sqlalchemy.create_engine`.
+        :return: the created engine.
+        """
+        connection = self.get_connection(self.gcp_conn_id)
+        credentials_path = json.loads(connection.extra)[
```

Comment:
``extra__google_cloud_platform__key_path`` is n optional key. Besides, we should also support other methods of providing credentials, including `extra__google_cloud_platform__keyfile_dict`, `extra__google_cloud_platform__key_secret_name`

Reply:
yea good catch, on it~

Reply:
@mik-laj now I implemented `keyfile_dict` + `export GOOGLE_APPLICATION_CREDENTIALS`

but `bigquery-sqlalchemy` doesn't support google-auth for now. If you get a chance you can refer to this issue for details 😂 
https://github.com/googleapis/python-bigquery-sqlalchemy/issues/42